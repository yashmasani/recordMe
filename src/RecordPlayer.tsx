import React,{useState, FunctionComponent} from 'react';
import {useReactMediaRecorder} from 'react-media-recorder';
import RecordButtonIcon from './assets/record-button-svgrepo-com.svg';
import stopRecordButtonIcon from './assets/stop-svgrepo-com.svg';

interface reactMediaRecorder{
  
    status: string,
    startRecording: ()=>void,
    stopRecording: ()=>void ,
    mediaBlobUrl: string | null


}


interface recorderProps {
    recordStart: boolean,
    setRecord: (recordStarted: boolean)=>void
}

export const RecordPlayer = ({recordStart, setRecord}: recorderProps ) =>{
  
  const {status, startRecording, stopRecording, mediaBlobUrl}:reactMediaRecorder = useReactMediaRecorder({video:false})
  
  
  
  const startClicked = () => {
      startRecording()
      setRecord(true)
  }

  const stopClicked = () => {
      stopRecording()
      setRecord(false)
  }

  

  return(
    <div className="" aria-label="Record Player">
        <button onClick={startClicked} aria-label="start Recording" disabled={recordStart}>:
            <img alt="Record button" src={RecordButtonIcon} className="player-buttons"/>
        </button>
        <button aria-label="stop Recording" onClick={stopClicked} disabled={!recordStart}>            <img alt="Stop Record Button" src={stopRecordButtonIcon} className="player-buttons"/> 
        </button>
    </div>
  )

}
