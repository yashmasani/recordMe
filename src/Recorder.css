@keyframes loadrecord {
    0% { transform: scale(0); opacity: 1;}
    100% { transform: scale(1); opacity: 0;}
}

.record-no-load {
    width:10rem;
    height:10rem;
    display: inline-block;
    position:relative;
    border: 2px solid black;
    border-radius: 50%;
}

.record-loader {
    width : 10rem;
    height: 10rem;
    display: inline-block;
    position: relative;
}

.record-loader::after , .record-loader::before {
    content: "";
    width : 10rem;
    height: 10rem;
    border-radius: 50%; 
    border:2px solid black;
    position: absolute;
    left:0;
    top: 0;
    animation:loadrecord 2s linear infinite;
}    

.record-loader::after {
    animation-delay: 1s;
} 
                                                                                                                                               
