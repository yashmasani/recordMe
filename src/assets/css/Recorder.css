@keyframes loadrecord {
    0% { transform: scale(0); opacity: 1;}
    100% { transform: scale(1); opacity: 0;}
}

.recorder{
    margin: 10%;
}

.record-no-load {
    width:10em;
    height:10em;
    display: inline-block;
    position:relative;
    border-radius: 50%;
    background-image: url('../svg/microphone-svgrepo-com.svg');
    background-size: 50%;
    background-repeat: no-repeat;
    background-position: center;
}

.record-loader {
    width : 10rem;
    height: 10rem;
    display: inline-block;
    position: relative;
}

.record-loader::after , .record-loader::before {
    content: "";
    width : 10rem;
    height: 10rem;
    border-radius: 50%; 
    border:2px solid white;
    position: absolute;
    left:0;
    top: 0;
    animation:loadrecord 2s linear infinite;
}    

.record-loader::after {
    animation-delay: 1s;
} 

.record-player {
    display:flex;
    flex-wrap: nowrap;
    justify-content: center;
}

.white-font{
    color:white
}

button>img:hover{
  background-color:white;
  transition: all 0.5s ease;
}

                                                                                                                                               
