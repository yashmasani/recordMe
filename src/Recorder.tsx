import React,{useState, FunctionComponent} from 'react';
import {RecordPlayer} from './RecordPlayer';
import './Recorder.css';

export const Recorder:FunctionComponent =  () => {
    
  
  const [recordStarted, setRecordStarted] = useState(false)

  return(
    <div>
      {/** Progress Bar with Animation  -- keeps moving until stop button  **/}
      <div className={recordStarted ? "record-loader" : "record-no-load"} aria-label="recording in progress loader"></div>
      <RecordPlayer  recordStart={recordStarted} setRecord={setRecordStarted} />
    </div>
    
  )

}

