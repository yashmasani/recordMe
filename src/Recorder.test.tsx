import React from 'react';
import '@testing-library/jest-dom';
import {Recorder} from './Recorder';
import renderer from 'react-test-renderer'
import {RecordPlayer} from './RecordPlayer';


  let mockSetRecord = jest.fn(()=>{})
  let  mockSetBlobUrl = jest.fn(()=>"") 
  let defaultStateRecord = false;

it('Recorder renders correctly at start', () => {
  
  const tree = renderer
    .create(<Recorder setRecord={mockSetRecord} setBlobUrl={mockSetBlobUrl} recordStart={defaultStateRecord}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/**
it('Idle state is rendered with start click', ()=>{

    const tree = renderer
    .create(<Recorder setRecord={mockSetRecord} setBlobUrl={mockSetBlobUrl} recordStart={defaultStateRecord}/>)
    
    const treeInstance = tree.root
    console.log(treeInstance.children)
    //expect(treeInstance.findByType()).toBe('Click Start to start Recording')


})
**/
