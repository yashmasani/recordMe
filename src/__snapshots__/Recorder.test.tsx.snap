// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Record Player renders correctly at start 1`] = `
<div
  aria-label="Record Player"
  className="record-player"
>
  <button
    aria-label="start Recording"
    disabled={false}
    onClick={[Function]}
  >
    <img
      alt="Record button"
      className="player-buttons round-border"
      src="record-button-svgrepo-com.svg"
    />
    <p
      className="white"
    >
      Start
    </p>
  </button>
  <button
    aria-label="stop Recording"
    disabled={true}
    onClick={[Function]}
  >
    <img
      alt="Stop Record Button"
      className="player-buttons"
      src="stop-svgrepo-com.svg"
    />
    <p>
      Stop
    </p>
  </button>
</div>
`;

exports[`Recorder renders correctly at start 1`] = `
<div
  className="recorder"
>
  <div
    aria-label="recording in progress loader"
    className="record-no-load"
  />
  <p>
    Click Start to start Recording
  </p>
  <div
    aria-label="Record Player"
    className="record-player"
  >
    <button
      aria-label="start Recording"
      disabled={false}
      onClick={[Function]}
    >
      <img
        alt="Record button"
        className="player-buttons round-border"
        src="record-button-svgrepo-com.svg"
      />
      <p
        className="white"
      >
        Start
      </p>
    </button>
    <button
      aria-label="stop Recording"
      disabled={true}
      onClick={[Function]}
    >
      <img
        alt="Stop Record Button"
        className="player-buttons"
        src="stop-svgrepo-com.svg"
      />
      <p>
        Stop
      </p>
    </button>
  </div>
</div>
`;
